sound/soc/sof/intel/hda-bus.c

```diff
@@ -90,7 +110,7 @@ void sof_hda_bus_init(struct hdac_bus *bus, struct device *dev,
 
 	bus->irq = -1;
 	bus->ext_ops = ext_ops;
-	bus->idx = idx++;
+	bus->idx = get_free_index();
```

Comment:
I don't get what the idx++ did in the first place, and how you remove it. 
What is the requirement and expected behavior for this idx?